### DISCUSSION TITLE
How to Return Unit Type from a Mutation in async-graphql?

### DISCUSSION THREAD

**Discussion by cosmir17** [async-graphql/async-graphql#1610] [Q&A]
Hello async-graphql team,

I'm working on a GraphQL API using `async-graphql` in Rust, and I'm facing an issue when trying to return a `Unit` type (`()`) from a mutation.

Here's the code for my mutation:

```rust
#[Object]
impl<W> Mutation<W>
where
......
{
    async fn my_method(&self, cx: &Context<'_>, id: String) -> Result<()> {
        let my_service = cx.get_service::<W>()?;
        my_service
            .do_this(&session_id.0)
            .await
            .inspect_err(|error| error!(error = ?error, "SERVICE ERROR"))?;
        Ok(())
    }
}
```

When I compile this code, I get the following error:
```
error[E0277]: the trait bound `(): OutputType` is not satisfied
  --> src/mutation.rs:46:1
   |
46 | #[Object]
   | ^^^^^^^^^ the trait `OutputType` is not implemented for `()`
   |
   = help: the following other types implement trait `OutputType`:
             &'a [T]
             &T
             Arc<T>
             Arc<[T]>
             BTreeMap<K, V>
             BTreeSet<T>
             Box<T>
             Box<[T]>
           and 61 others
   = note: this error originates in the attribute macro `Object` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I understand that () (unit type) does not implement OutputType, which is required by async-graphql for return types in mutations.

My goal is to have the following GraphQL schema:
```
type Mutation {
  my_method(id: String!): Unit!
}
```

But currently, the schema generated doesn't work due to the compile error.

I couldn't find a way to return Unit in the async-graphql documentation or examples. Is there a way to define a Unit type in async-graphql so that I can return it from my mutation?

Any guidance or examples would be greatly appreciated!

Thank you!

**Comment by sunli829** [comment]
No, the GraphQL specification does not support mutations without a return value.

**Reply by cosmir17** [reply]
Thank you, @sunli829 for your answer. 
The original project I am migrating from, it's schema file has 'Unit' scalar type.

I didn't know `the GraphQL specification does not support mutations without a return value.`. Hence, they created the Unit type in the original project.

I missed this before I raised this question.
